testslide/strict_mock.py

```diff
@@ -623,8 +633,8 @@ def __validate_attribute_type(self, name: str, value: Any) -> None:
         ):
             return
 
-        if hasattr(self._template, "__annotations__"):
-            annotations = self._template.__annotations__
+        if self._template is not None:
+            annotations = get_type_hints(self._template)
```

Comment:
This does not play nice with pybind11 based modules
```
in get_type_hints
    base_globals = sys.modules[base.__module__].__dict__
KeyError: 'pybind11_builtins'
```

Reply:
Thanks, this seems to have caused an issue in sphinx as well where they catch keyerror and have empty annotations. I don't have the right setup but it will be good to know what was the value of annotations using `__annotations__` attribute in earlier versions.

https://github.com/sphinx-doc/sphinx/issues/8084

Reply:
I guess catching key error and returning empty annotations would work well as a workaround for us as well

Reply:
Thanks, I will make the changes. On further search seems this is an open issue in CPython where there is a PR open to catch the KeyError and discard it for these type of cases : https://bugs.python.org/issue41515

Reply:
https://www.python.org/dev/peps/pep-0563/#runtime-annotation-resolution-and-class-decorators it can throw NameError on metaclasses